capstone/src/main/webapp/scripts/poll.js

```diff
@@ -26,16 +27,49 @@ function getPollOptions() {
       const optionsContainer = document.getElementById("options-container");
       optionsContainer.innerHTML = "";
       let maxVotes = getMaxVotes(pollData);
+      topChallenge = pollData["options"][0]["text"];
       pollData["options"].forEach(option => {
         renderOptionElement(option, maxVotes);
       });
       return pollData["votedOptions"];
     })
     .then(votedOptions => {
       handleCheck(votedOptions);
+    })
+    .then(checkWeek);
+}
+
+function checkWeek() {
+  getChallenge();
+  let now = new Date();
+  //Hard coding due date for now
+  let millisTillDueDate =
+    new Date(now.getFullYear(), now.getMonth(), now.getDate(), 21, 8, 0, 0) -
+    now;
+  if (millisTillDueDate >= 0) {
+    setTimeout(updatePoll, millisTillDueDate);
+  }
```

Comment:
I might be misunderstanding here but if millisTillDueDate was negative, shouldn't you manually call updatePoll(), because that indicated the current challenge is outdated?

Reply:
yeah I still have to write in code that checks how many weeks have passed and update the new challenge accordingly